### PR TITLE
feat: `x/blocksdk` inital proto

### PR DESCRIPTION
**Pull request by aljo242**
Targetting `blocksdkmodule` branch.

- add `lane.proto` and `module.proto`
- run `make proto-gen`, `make-proto-pulsar-gen`, `make proto-format`

### CODE REVIEW

File: proto/sdk/blocksdk/v1/blocksdk.proto [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    (gogoproto.customtype) = "cosmossdk.io/math.LegacyDec",
+    (amino.dont_omitempty) = true,
+    (gogoproto.nullable) = false
+  ];
+
+  // order is the priority ordering of the Lane.  Lane orders should be set in
+  // order of priority starting from 0, monotonically increasing and
+  // non-overlapping.  A lane with a lower order value will have a higher
+  // priority over a lane with a higher order value.  For example, if LaneA has
+  // priority of 0 and LaneB has a priority of 1, LaneA has priority over LaneB.
```
[the remark this answers is not in the document]
Comment (aljo242) [reply proto/sdk/blocksdk/v1/blocksdk.proto:34]: Are those two things different?  I was grouping them together.  Is there a better term used to disinguish?

[the remark this answers is not in the document]
Comment (nivasan1) [reply proto/sdk/blocksdk/v1/blocksdk.proto:34]: nah, I j was hoping to get clarification. Would potentially be useful to specify that ordering of lanes is specifically used in Prepare / ProcessProposal ordering

[the remark this answers is not in the document]
Comment (nivasan1) [reply proto/sdk/blocksdk/v1/blocksdk.proto:34]: Otherwise lgtm, merge whenever

[the remark this answers is not in the document]
Comment (aljo242) [reply proto/sdk/blocksdk/v1/blocksdk.proto]: https://github.com/skip-mev/block-sdk/pull/183/commits/27d774834a87ca53a01589f5e102ce8b42460960

**Pull request merged**